runelite-client/src/main/java/net/runelite/client/plugins/deathindicator/DeathIndicatorPlugin.java

```diff
@@ -0,0 +1,311 @@
+					deathTimer = new Timer(Duration.ofHours(1).minus(Duration.between(Instant.parse(config.timeOfDeath()), Instant.now())).getSeconds(), ChronoUnit.SECONDS, BONES, this);
+					deathTimer.setTooltip("Died on world: " + config.deathWorld());
+					infoBoxManager.addInfoBox(deathTimer);
+				}
+			}
+		}
+	}
+
+	@Subscribe
+	public void onGameTick(GameTick event)
+	{
+		if (!config.hasDied() || !config.hasRespawned() || (client.getWorld() != config.deathWorld()))
+		{
+			return;
+		}
+
+		if ((client.getLocalPlayer().getWorldLocation().getX() == config.deathLocationX())
+			&& (client.getLocalPlayer().getWorldLocation().getY() == config.deathLocationY())
+			&& (client.getLocalPlayer().getWorldLocation().getPlane() == config.deathLocationPlane())
+			|| ((deathTimer != null) && (deathTimer.getText().equals("0:00"))))
+		{
+			if (client.hasHintArrow())
+			{
+				client.clearHintArrow();
+			}
+
+			if (deathTimer != null)
+			{
+				infoBoxManager.removeInfoBox(deathTimer);
+				deathTimer = null;
+			}
+
+			worldMapPointManager.removeIf(DeathWorldMapPoint.class::isInstance);
+			config.deathMarkerReset(true);
+		}
+
+		if (config.deathMarkerReset())
+		{
+			config.hasDied(false);
+			config.hasRespawned(true);
+			config.deathMarkerReset(false);
+			config.deathWorld(-1);
+			config.deathLocationX(-1);
+			config.deathLocationY(-1);
+			config.deathLocationPlane(-1);
+			config.timeOfDeath("");
+		}
+	}
+
+	@Subscribe
+	public void onConfigChanged(ConfigChanged event)
+	{
+		if (event.getGroup().equals("deathIndicator"))
+		{
+			if (!config.showDeathHintArrow() && client.hasHintArrow())
+			{
+				client.clearHintArrow();
+			}
+
+			if (!config.showDeathInfoBox() && deathTimer != null)
+			{
+				infoBoxManager.removeInfoBox(deathTimer);
+				deathTimer = null;
+			}
+
+			if (!config.showDeathOnWorldMap())
+			{
+				worldMapPointManager.removeIf(DeathWorldMapPoint.class::isInstance);
+			}
+
+			// Simulates hitting the reset button, removes death info!
+			if ((config.deathLocationX() == -1)
+				&& (config.deathLocationY() == -1)
+				&& (config.deathLocationPlane() == -1)
+				&& (!config.hasDied())
+				&& (config.hasRespawned())
+				&& (!config.deathMarkerReset())
+				&& (config.deathWorld() == -1)
+				&& (config.timeOfDeath().equals("")))
+			{
+				if (client.hasHintArrow())
+				{
+					client.clearHintArrow();
+				}
+
+				if (deathTimer != null)
+				{
+					infoBoxManager.removeInfoBox(deathTimer);
+					deathTimer = null;
+				}
+
+				worldMapPointManager.removeIf(DeathWorldMapPoint.class::isInstance);
+			}
+		}
+	}
+
+	@Subscribe
+	public void onGameStateChanged(GameStateChanged event)
+	{
+		if (!config.hasDied())
+		{
+			return;
+		}
+
+		if (event.getGameState() == GameState.LOGGED_IN)
+		{
+			if (client.getWorld() == config.deathWorld())
+			{
+				if (config.showDeathHintArrow())
+				{
+					client.setHintArrow(new WorldPoint(config.deathLocationX(), config.deathLocationY(), config.deathLocationPlane()));
```

Comment:
This arrow will get deleted when you f.e get bounty hunter target or do quests/clues etc (same with the onActorDeath)

Reply:
I'm not sure exactly what to do to fix that issue, can you elaborate a bit more? Thanks :)

Reply:
So @deathbeam , should I just do something like:

if (client.hasHintArrow())
{
return;
}
else
{
client.setHintArrow(...);
}

That way if a different arrow already exists I don't override it/clear it?